File: intellij-plugin/src/main/kotlin/com/ezevillo/gitreview/domain/SoleTarget.kt
package com.ezevillo.gitreview.domain

/**
 * Single usable root, like CLI cwd: 0 → none; 1 → that; 2+ → none (no guessing).
 */
fun <T> pickSoleTarget(targets: List<T>): T? =
    if (targets.size == 1) targets[0] else null

File: visualstudio-extension/src/GitReview.Domain/SoleTarget.cs
namespace GitReview.Domain;

public static class SoleTarget
{
    /// <summary>Single usable root, like CLI cwd: 0 → none; 1 → that; 2+ → none (no guessing).</summary>
    public static T? PickSoleTarget<T>(IReadOnlyList<T> targets) =>
        targets.Count == 1 ? targets[0] : default;
}
